charts/shoot-core/components/charts/kube-proxy/templates/_helpers.tpl

```diff
@@ -1,3 +1,11 @@
+{{- define "kube-proxy.name" -}}
+{{- if eq .name "" -}}
+kube-proxy
+{{- else -}}
+kube-proxy-{{ .name }}-v{{ .kubernetesVersion }}
```

Comment:
Do we really need the `kubernetesVersion` in the name? While it arguably makes more evident which k8s version is this deamonset using, it does make the name longer. Perhaps it would be better to omit it?
```suggestion
kube-proxy-{{ .name }}
```

Reply:
Without the version in the name there would be only one `DaemonSet` per worker pool. This way we would end up with similar problems like in the `master` branch: When a Kubernetes patch version update is performed then the node selector if the `DaemonSet` is already updated to the new patch version while the `Node`s might still be labeled with the old version. It can take up to `5m` until the label is updated/the kubelet is restarted. During this time frame, `kube-proxy` would be deleted since its node selector was already changed.

Reply:
OK, it's clear now that you have explained how we could get to this situation in the `master` branch, I didn't get this completely before. Thanks!